src/components/pages/item/type.ts

```diff
@@ -0,0 +1,6 @@
+import { DiscriminatedItem, PermissionLevel } from '@graasp/sdk';
+
+export interface OutletType {
+  item: DiscriminatedItem;
+  permission?: PermissionLevel;
```

Comment:
Looking closely at this, I think that the optional can be avoided if the computations for the `permission` is moved inside the if in `ItemScreenLayout`. We should still ensure that access is allowed when public though. But this would simplify the logic in some places.

Reply:
The membership could be null, so the permission might not exist. I think you still want to proceed if the item is public, so well, permission can be `undefined`.

Reply:
Okay sounds good